src/murmur/Messages.cpp

```diff
@@ -195,34 +195,48 @@ void Server::msgAuthenticate(ServerUser *uSource, MumbleProto::Authenticate &msg
 
 	bool ok     = false;
 	bool nameok = validateUserName(uSource->qsName);
-	QString pw  = u8(msg.password());
-
-	// Fetch ID and stored username.
-	// Since this may call DBus, which may recall our dbus messages, this function needs
-	// to support re-entrancy, and also to support the fact that sessions may go away.
-	int id = authenticate(uSource->qsName, pw, uSource->uiSession, uSource->qslEmail, uSource->qsHash,
-						  uSource->bVerified, uSource->peerCertificateChain());
-
-	uSource->iId = id >= 0 ? id : -1;
 
 	QString reason;
 	MumbleProto::Reject_RejectType rtType = MumbleProto::Reject_RejectType_None;
+	int id                                = -2;
+
+	if (!bForceUsernameCertSubjectEquality
+		|| (bCertRequired
+			&& uSource->qsName
+				   == uSource->peerCertificateChain()
+						  .first() // First in chain is the users certificate
```

Comment:
```suggestion
						  .first() // First in user's certificate chain.
```

Reply:
"First in chain is the users certificate" means that the `first` certificate in that chain (`uSource->peerCertificateChain()`) is the certificate that was issued to the user (while subsequent certificates in that chain are intermediates and finally the `last` one is a root certificate [note: root certificates might be omitted according to the standard]). I fear that this meaning is not transported by saying "First in user's certificate chain." Could we agree on saying "The first certificate is the one explicitly issued to the user, while subsequent certificates are intermediate certificates."? I know it's a bit long but it should make things clearer.